src/main/java/com/google/sps/data/Organization.java

```diff
@@ -47,4 +52,89 @@ public Organization(Entity entity) {
      this.urlLink = (String) entity.getProperty("orgUrl");
      this.moderators = (ArrayList) entity.getProperty("moderatorList");
   }
+
+  public void updateOrganization(HttpServletRequest request, boolean isMaintainer, boolean isModerator) throws IllegalArgumentException{
+    Map<String, String> properties = new HashMap<String,String>();
+
+    // Format is Form Name, Entity Property 
+    properties.put("org-name", "orgName");
+    properties.put("org-email", "orgEmail");
+    properties.put("org-street-address", "orgStreetAddress");
+    properties.put("org-phone-num", "orgPhoneNum");
+    properties.put("org-url", "orgUrl");
+    properties.put("org-description","orgDescription");
+    properties.put("approval", "isApproved");
+    properties.put("moderator-list", "moderatorList");
+
+    for(Map.Entry<String, String> entry : properties.entrySet()) {
+      String formKey = entry.getKey();
+      String propertyKey = entry.getValue();
+      String formValue = "";
+
+      // will only get approval param is a maintainer sent the request
+      if(propertyKey.equals("isApproved") && !isMaintainer) {
+        continue;
+      }
+
+      // will only get moderatorList param if request is sent by a moderator or maintainer
+      if(propertyKey.equals("moderatorList") && !isModerator && !isMaintainer) {
+        continue;
+      }
+
+      // setting organization description is optional
+      if(!propertyKey.equals("orgDescription")) {
+        try {
+          formValue = getParameterOrThrow(request, formKey);
+        } catch(IllegalArgumentException err) {
+          throw new IllegalArgumentException();
+        }
+      } else {
+        formValue = request.getParameter(formKey);
+      }
+      setOrganizationProperty(propertyKey, formValue);
+    }
+  }
+
+  public void commitOrganization(Entity entity) {
+    entity.setProperty("orgName", name);
+    entity.setProperty("orgEmail", email);
+    entity.setProperty("orgStreetAddress", address);
+    entity.setProperty("orgDescription", description);
+    entity.setProperty("orgPhoneNum", phoneNum);
+    entity.setProperty("isApproved", isApproved);
+    entity.setProperty("orgUrl", urlLink);
+    entity.setProperty("moderatorList", moderators);
+  }
+
+  private String getParameterOrThrow(HttpServletRequest request, String formKey) {
+    String result = request.getParameter(formKey);
+    if(result.isEmpty()) {
+        throw new IllegalArgumentException("Form value cannot be null");
+    }
+    return result;
+  }
+
+  private void setOrganizationProperty(String propertyKey, String formValue) {
+    if(propertyKey.equals("moderatorList")) {
```

Comment:
If you store the entity, this can then instead be:

this.entity_.setProperty(propertyKey, formValue);

for everything except moderatorList and isApproved

ie;

if (propertyKey is moderatorList) {
  // what you have now
} else if (propertyKey is apprived) {
  // what you have now
} else {
  // this handles everything else.
  this.entity_.setProperty(propertyKey, formValue);
}